manifestservice/manifests/__init__.py

```diff
@@ -0,0 +1,259 @@
+    We find the appropriate folder ("prefix") in the bucket by asking Fence for 
+    info about the user's access token.
+    ---
+    responses:
+        200:
+            description: Success
+        403:
+            description: Unauthorized
+    """
+    
+    try:
+        set_current_token(validate_request(aud={"user"}))
+    except Exception:
+        json_to_return = { "error" : "Please log in." }
+        return flask.jsonify(json_to_return), 401
+    
+    auth_successful = does_the_user_have_read_access_on_at_least_one_project(current_token)
+
+    if not auth_successful:
+        json_to_return = { "error" : "You must have read access on at least one project in order to use this feature." }
+        return flask.jsonify(json_to_return), 403
+
+    folder_name = get_folder_name_from_token(current_token)
+
+    MANIFEST_BUCKET_NAME = flask.current_app.config.get("MANIFEST_BUCKET_NAME")
+    json_to_return = {
+        "manifests" : list_files_in_bucket(MANIFEST_BUCKET_NAME, folder_name)
+    }
+
+    return flask.jsonify(json_to_return), 200
+
+@blueprint.route("/file/<file_name>", methods=["GET"])
+def get_manifest_file(file_name):
+    """
+    Returns the requested manifest file from the user's folder.
+    ---    
+    responses:
+        200:
+            description: Success
+        403:
+            description: Unauthorized
+        400:
+            description: Bad request format
+    """
+
+    try:
+        set_current_token(validate_request(aud={"user"}))
+    except Exception:
+        json_to_return = { "error" : "Please log in." }
+        return flask.jsonify(json_to_return), 401
+    
+    auth_successful = does_the_user_have_read_access_on_at_least_one_project(current_token)
+
+    if not auth_successful:
+        json_to_return = { "error" : "You must have read access on at least one project in order to use this feature." }
+        return flask.jsonify(json_to_return), 403
+    
+    if not file_name.endswith("json"):
+        json_to_return = { "error" : "Incorrect usage. You can only use this pathway to request files of type JSON." }
+        return flask.jsonify(json_to_return), 400
+    
+    folder_name = get_folder_name_from_token(current_token)
+
+    MANIFEST_BUCKET_NAME = flask.current_app.config.get("MANIFEST_BUCKET_NAME")
+    json_to_return = {
+        "body" : get_file_contents(MANIFEST_BUCKET_NAME, folder_name, file_name)
+    }
+
+    return flask.jsonify(json_to_return), 200
+
+def add_manifest_to_bucket(current_token, manifest_json):
+    """
+    Puts the manifest_json string into a file and uploads it to s3. 
+    Generates and returns the name of the new file.
+    """
+    folder_name = get_folder_name_from_token(current_token)
+
+    MANIFEST_BUCKET_NAME = flask.current_app.config.get("MANIFEST_BUCKET_NAME")
+    filename = generate_unique_manifest_filename(folder_name, MANIFEST_BUCKET_NAME)
+    manifest_as_bytes = str.encode(str(flask.request.json))
```

Comment:
I would either pass the `flask.request.json` in as an arg, or make it clear that this has to be called in context of a flask request. Or just prefix the function with `_` so people know it's internal